sound/soc/sof/pm.c

```diff
@@ -303,23 +285,8 @@ int snd_sof_suspend(struct device *dev)
 	struct snd_sof_dev *sdev = dev_get_drvdata(dev);
 	int ret;
 
-	if (snd_sof_dsp_d0i3_on_suspend(sdev)) {
-		/* suspend to D0i3 */
-		dev_dbg(sdev->dev, "DSP is trying to enter D0i3...\n");
-		ret = snd_sof_set_dsp_power_state(sdev, SOF_DSP_D0I3);
-		if (ret == -ENOTSUPP) {
-			/* fallback to D3 suspend */
-			dev_dbg(sdev->dev, "D0i3 not supported, fall back to D3...\n");
-			goto d3_suspend;
-		} else if (ret < 0) {
-			dev_err(sdev->dev, "error: failed to enter D0I3, %d\n",
-				ret);
-			return ret;
-		}
-
-		/* platform-specific suspend to D0i3 */
+	if (snd_sof_dsp_d0i3_on_suspend(sdev))
 		return snd_sof_dsp_suspend(sdev, SOF_DSP_D0I3);
```

Comment:
symmetrical on suspend and resume?

Reply:
it isnt symmertic and there's no justification to make it symmetric. During suspend, we need to determing whether to enter D0i3 or D3. During resme, we always resume to D0.

Reply:
@ranj063 I meant even we always resuming to D0, but resuming from D0I3 and D3 are different. I can understand it, but the dissymmetry might make difficulty for other reviewers who are not familiar with the details to follow why you do this specific in suspend but don't do the corresponding in resume.

Reply:
@keyonjie BTW I forgot to mention that this is changed in the subsequent patches where it does become sort of symmetric